controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentOrder.java

```diff
@@ -88,11 +93,23 @@ public boolean isLast(JobType job, Application application) {
     /** Returns jobs for given deployment spec, in the order they are declared */
     public List<JobType> jobsFrom(DeploymentSpec deploymentSpec) {
         return deploymentSpec.steps().stream()
-                .flatMap(step -> step.zones().stream())
+                .flatMap(step -> jobsFrom(step).stream())
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+    }
+
+    /** Returns jobs for the given step */
+    private List<JobType> jobsFrom(DeploymentSpec.Step step) {
+        return step.zones().stream()
                 .map(this::toJob)
                 .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
     }
 
+    /** Returns whether given step has deployed completely */
+    private boolean isDeployed(DeploymentSpec.Step step, Application application) {
```

Comment:
Same here, should be names "isSuccessful", or maybe even "completedSuccessfully" imho.

Reply:
Changed, thanks.